gtsfm/evaluation/metrics.py

```diff
@@ -0,0 +1,287 @@
+"""Class to store metrics computed in different GTSfM modules.
+
+Authors: Akshay Krishnan
+"""
+from __future__ import annotations
+
+import json
+import numpy as np
+from enum import Enum
+from typing import Any, Dict, List, Union
+
+import gtsfm.utils.io as io
+
+DATA_KEY = "full_data"
+SUMMARY_KEY = "summary"
+
+
+class GtsfmMetric:
+    """Class to store a metric computed in a GTSfM module."""
+
+    class PlotType(Enum):
+        BAR = 1  # For scalars
+        BOX = 2  # For 1D distributions
+        HISTOGRAM = 3  # For 1D distributions
+
+    def _get_plot_types_for_dim(self, dim) -> List[PlotType]:
```

Comment:
Thanks for the PR, Akshay. Looks great. Missing type hint on dim

Reply:
Double underscore for private methods?

Reply:
Looks like single underscores are the recommended choice, do we have a gtsfm specific convention for this? https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles